tests/gordo_components/cli/test_cli.py

```diff
@@ -202,3 +207,124 @@ def test_expand_model_complains_on_missing_vars(self):
         model_template = "{'gordo_components.model.models.KerasAutoEncoder': {'kind': '{{kind}}', 'num': {{num}}}} "
         with self.assertRaises(ValueError):
             expand_model(model_template, model_params)
+
+
+@pytest.mark.parametrize(
+    "should_be_equal, cv_mode", [(True, "full_build"), (False, "cross_val_only")]
+)
+def test_build_cv_mode(should_be_equal: bool, cv_mode: str):
+    """
+    Testing build with cv_mode set to full and cross_val_only. Checks that cv_scores are
+    printed and model are only saved when using the default (full) value.
+    """
+
+    runner = CliRunner()
+
+    logger.info(f"MODEL_CONFIG={json.dumps(MODEL_CONFIG_WITH_PREDICT)}")
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        with temp_env_vars(
+            MODEL_NAME="model-name",
+            OUTPUT_DIR=tmpdir,
+            DATA_CONFIG=DATA_CONFIG,
+            MODEL_CONFIG=json.dumps(MODEL_CONFIG_WITH_PREDICT),
+        ):
+            location_file = f"{os.path.join(tmpdir, 'location.txt')}"
+            result = runner.invoke(
+                cli.gordo,
+                [
+                    "build",
+                    "--print-cv-scores",
+                    f"--cv-mode={cv_mode}",
+                    f"--model-location-file={location_file}",
+                ],
+            )
+            # Checks that the file is empty or not depending on the mode.
+            # Also checks the output contains 'explained-variance_raw-scores'
+            if should_be_equal:
+                assert os.stat(location_file).st_size != 0
+                assert "explained-variance_raw-scores=" in result.output
+            else:
+                assert os.stat(location_file).st_size == 0
+                assert "explained-variance_raw-scores=" in result.output
+
+
+@pytest.mark.parametrize(
+    "should_be_equal,cv_mode_1, cv_mode_2",
+    [
+        (True, "full_build", "cross_val_only"),
+        (False, "cross_val_only", "cross_val_only"),
+    ],
+)
+def test_build_cv_mode_cross_val_cache(
+    should_be_equal: bool, cv_mode_1: str, cv_mode_2: str
+):
+    """
+    Checks that cv_scores uses cache if runned after a full build. Loads the same model, and can 
+    print the cv_scores from them. 
+    """
+
+    runner = CliRunner()
+
+    logger.info(f"MODEL_CONFIG={json.dumps(MODEL_CONFIG)}")
+
+    with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
Could save yourself a few lines in these tests by using the `tmp_dir` fixture if you like.
https://github.com/equinor/gordo-components/blob/63c53ef9eade25732f683f757adb9a3ee5ce193b/tests/conftest.py#L49